.github/workflows/ci/ui.yaml

```diff
@@ -0,0 +1,77 @@
+name: ci/ui
+
+on:
+  pull_request:
+    paths:
+      - 'ui/**'
+  push:
+    branches:
+      - main
+    paths:
+      - 'ui/**'
+
+defaults:
+  run:
+    working-directory: ui
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+          cache-dependency-path: ui/package-lock.json
+      - uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: ${{ runner.os }}-npm-
+      - name: Dependencies
+        run: npm install
+      - name: Lint
+        run: npm run lint
+      - name: Format
+        run: npm run format
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: npm
+          cache-dependency-path: ui/package-lock.json
+      - uses: actions/cache@v3
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-npm-${{ hashFiles('package-lock.json') }}
+          restore-keys: ${{ runner.os }}-npm-
+      - name: Dependencies
+        run: npm install
```

Comment:
Not necessary to be part of this PR, but I wonder if we can run all these tests on the same image so that we don't have to `npm install` 3 separate times, which looks like it may re-install on the dependencies on a fresh image each time

Reply:
That's a valid point. The cache step above should help alleviate some of those concerns